dask/dask.sh

```diff
@@ -0,0 +1,242 @@
+#!/bin/bash
+
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This initialization action script will install Dask and other relevant
+# libraries on a Dataproc cluster. This is supported for either "yarn" or
+# "standalone" runtimes Please see dask.org and yarn.dask.org for more
+# information.
+
+set -euxo pipefail
+
+if (which anaconda); then
+  CONDA_MANAGER=anaconda
+else
+  CONDA_MANAGER=miniconda3
+fi
+readonly CONDA_MANAGER
+
+readonly DASK_CONFIG_DIR=/etc/dask/
+readonly DASK_CONFIG_FILE=${DASK_CONFIG_DIR}/config.yaml
+
+readonly DASK_ENV_DIR=/opt/envs
+readonly DASK_ENV_NAME=dask
+readonly DASK_ENV_PACK=${DASK_ENV_DIR}/${DASK_ENV_NAME}.tar.gz
+readonly DASK_ENV=/opt/conda/${CONDA_MANAGER}/envs/${DASK_ENV_NAME}
+
+readonly DASK_RUNTIME="$(/usr/share/google/get_metadata_value attributes/dask-runtime || echo "yarn")"
+readonly RUN_WORKER_ON_MASTER="$(/usr/share/google/get_metadata_value attributes/dask-worker-on-master || echo "true")"
+
+readonly CONDA_EXTRA_PACKAGES="$(/usr/share/google/get_metadata_value attributes/CONDA_PACKAGES || echo "")"
+readonly CONDA_EXTRA_CHANNELS="$(/usr/share/google/get_metadata_value attributes/CONDA_CHANNELS || echo "")"
+
+readonly ROLE=$(/usr/share/google/get_metadata_value attributes/dataproc-role)
+readonly MASTER=$(/usr/share/google/get_metadata_value attributes/dataproc-master)
+
+# Dask 'standalone' config
+readonly DASK_LAUNCHER='/usr/local/bin/dask-launcher.sh'
+readonly DASK_SERVICE='dask-cluster'
+
+mkdir -p ${DASK_ENV_DIR} ${DASK_CONFIG_DIR}
+
+readonly CONDA_BASE_CHANNELS=(
+    "conda-forge"
+)
+
+CONDA_BASE_PACKAGES=(
+    "dask>=2.6.0"
+    "dill"
+    "ipykernel"
+    "fastparquet"
+    "gcsfs"
+    "pyarrow"
+)
+
+if [[ "${DASK_RUNTIME}" == "yarn" ]]; then
+  CONDA_BASE_PACKAGES+=("dask-yarn~=0.8")
```

Comment:
This version condition will install `dask-yarn` packages greater than 0.8 version but smaller than 1.0. Is this intentional?

Reply:
Some of the features of `dask-yarn` are experimental, so I think re-evaluating it upon a major version update is a good idea. What would you suggest for this use case, if not this?

Reply:
If we are pinning package then we should pin major.minor version, so we can pin it to 0.8 version for Dataproc 1.5: `dask-yarn=0.8`

For 2.0 preview version we can unpin it until it in preview

Reply:
Fixed